### PR TITLE
ARUHA-817 Validate occured_at in a more accurate way

### PR DESCRIPTION
**Pull request by antban**
ARUHA-817

### CODE REVIEW

**Comment by rcillo** [comment]
👍

**Comment by antban** [comment]
:+1:

**Comment by zrobgar** [comment]
👍

**Comment by antban** [comment]
:+1:

File: src/main/java/org/zalando/nakadi/validation/RFC3339DateTimeValidator.java [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
-import static java.time.format.DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+import org.everit.json.schema.FormatValidator;
 
 public class RFC3339DateTimeValidator implements FormatValidator {
 
     private final String errorMessage = "must be a valid date-time";
 
     // Valid offsets are either Z or hh:mm. The format hh:mm:ss is not valid
-    private final String dateTimeOffsetPattern = "^.*(Z|((\\+|-)\\d\\d:\\d\\d))$";
+    private final String dateTimeOffsetPattern = "^.*[Tt]\\d{2}:\\d{2}:\\d{2}.*([zZ]|([+-]\\d{2}:\\d{2}))$";
```
Comment (adyach) [inline src/main/java/org/zalando/nakadi/validation/RFC3339DateTimeValidator.java]: it matches to 1996-10-15T16:39:57.123+**99:99**

Comment (antban) [inline src/main/java/org/zalando/nakadi/validation/RFC3339DateTimeValidator.java]: @adyach It won't be accepted because of `OffsetDateTime.parse`

**Comment by adyach** [comment]
👍

**Comment by antban** [comment]
:+1:

File: src/main/java/org/zalando/nakadi/validation/RFC3339DateTimeValidator.java (hunk 1 above)
Comment (ePaul) [inline src/main/java/org/zalando/nakadi/validation/RFC3339DateTimeValidator.java]: Do you really want a `.*` here for the fractional seconds?

Comment (rcillo) [inline src/main/java/org/zalando/nakadi/validation/RFC3339DateTimeValidator.java]: Even if it's improved to be just numbers, this is also covered by the Java `OffsetDateTime.parse`.

Unfortunately, this parser is not as strict as we would like, so we need to have this additional regex to cover corner cases.

Comment (ePaul) [inline src/main/java/org/zalando/nakadi/validation/RFC3339DateTimeValidator.java]: Hmm, maybe indicate this purpose in the comment above, like `// regular expression to catch validation cases not caught by OffsetDateTime.parse`.

**Comment by antban** [comment]
:+1:

**Comment by adyach** [comment]
👍

**Comment by antban** [comment]
:+1:

**Comment by antban** [comment]
:+1:

**Comment by lmontrieux** [comment]
👍

**Comment by antban** [comment]
:+1:

**Pull request merged**